src/Microsoft.Framework.FileSystemGlobbing/Matcher.cs

```diff
@@ -10,24 +10,24 @@ namespace Microsoft.Framework.FileSystemGlobbing
 {
     public class Matcher
     {
-        private IList<IPattern> _includePatterns = new List<IPattern>();
-        private IList<IPattern> _excludePatterns = new List<IPattern>();
+        protected IList<IPattern> IncludePatterns { get; set; } = new List<IPattern>();
+        protected IList<IPattern> ExcludePatterns { get; set; } = new List<IPattern>();
```

Comment:
The collection is not essential to external consumption. If it is public, then we need to guard the situations in which a invalid pattern is added to the pattern collection.